src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -327,17 +329,27 @@ def __init__(self, mgr, cluster_id):
         self.mgr = mgr
 
     def create_empty_rados_obj(self):
-        common_conf = 'conf-nfs'
+        common_conf = 'conf-nfs.{}'.format(self.cluster_id)
         result = ''
         with self.mgr.rados.open_ioctx(self.pool_name) as ioctx:
             if self.pool_ns:
                 ioctx.set_namespace(self.pool_ns)
             ioctx.write_full(common_conf, result.encode('utf-8'))
             log.debug(
-                    "write configuration into rados object %s/%s/nfs-conf\n",
-                    self.pool_name, self.pool_ns)
-
-    def create_nfs_cluster(self, export_type, size):
+                    "write configuration into rados object %s/%s/%s\n",
+                    self.pool_name, self.pool_ns, common_conf)
+
+    def check_cluster_exists(self):
+        try:
+            completion = self.mgr.describe_service(service_type='nfs')
+            self.mgr._orchestrator_wait([completion])
+            orchestrator.raise_if_exception(completion)
+            return self.cluster_id in [cluster.spec.service_id for cluster in completion.result]
+        except Exception as e:
+            log.exception(str(e))
+            return True
+
+    def create_nfs_cluster(self, export_type, size, hosts):
```

Comment:
```suggestion
    def create_nfs_cluster(self, export_type, placement_spec):
```

by limiting this to size + hosts, you'll make it impossible to use labels for example. 

https://docs.ceph.com/docs/master/mgr/orchestrator/#placement-specification